catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/metacard/AbstractBasicMetacardUpdater.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.video.stream.mpegts.metacard;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+
+/**
+ * Copies child attribute to parent if a specific condition is met.
+ */
+public abstract class AbstractBasicMetacardUpdater implements MetacardUpdater {
+
+    private final String attributeName;
+
+    protected AbstractBasicMetacardUpdater(String attributeName) {
+        this.attributeName = attributeName;
+    }
+
+    @Override
+    public final void update(Metacard parent, Metacard child) {
+        if (isChildAvailable(child) && isCondition(parent, child)) {
+            parent.setAttribute(new AttributeImpl(attributeName,
```

Comment:
it was never a concern before, but I went ahead and added a check that the parent has an attribute descriptor for the attribute name being set.